Add test coverage reporting

nstuardod opened the pull request:
Se reemplaza el clásico `manage.py test` por `pytest` y se agrega test coverage en CircleCI.

nstuardod committed:
test: add pytest and test coverage

wooo- commented:
No están pasando los tests con pytest

<pre>&gt; pytest
<b>======================================== test session starts ========================================</b>
platform linux -- Python 3.8.10, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /home/wooo/.cache/pypoetry/virtualenvs/django-project-template-8zXZtD0D-py3.8/bin/python
cachedir: .pytest_cache
django: settings: project.settings (from ini)
rootdir: /mnt/magnet/repos/nstuardod-dpt, configfile: pytest.ini
plugins: cov-3.0.0, django-4.4.0
<b>collected 7 items                                                                                   </b>

base/tests.py::UrlsTest::test_responses <font color="#CC0000">FAILED                                                [ 14%]</font>
base/tests.py::CheckErrorPages::test_404 <font color="#4E9A06">PASSED</font><font color="#CC0000">                                               [ 28%]</font>
base/tests.py::CronTests::test_cron_classes_to_run <font color="#4E9A06">PASSED</font><font color="#CC0000">                                     [ 42%]</font>
parameters/tests.py::ParameterTestCase::test_create_all_parammeters <font color="#4E9A06">PASSED</font><font color="#CC0000">                    [ 57%]</font>
regions/tests.py::SimpleTest::test_basic_addition <font color="#4E9A06">PASSED</font><font color="#CC0000">                                      [ 71%]</font>
users/tests.py::UserTests::test_force_logout <font color="#4E9A06">PASSED</font><font color="#CC0000">                                           [ 85%]</font>
users/tests.py::UserTests::test_lower_case_emails <font color="#4E9A06">PASSED</font><font color="#CC0000">                                      [100%]</font>

============================================= FAILURES ==============================================
<font color="#CC0000"><b>______________________________________ UrlsTest.test_responses ______________________________________</b></font>
Traceback (most recent call last):
  File &quot;/mnt/magnet/repos/nstuardod-dpt/base/tests.py&quot;, line 238, in test_responses
    test_url_patterns(urlpatterns)
  File &quot;/mnt/magnet/repos/nstuardod-dpt/base/tests.py&quot;, line 236, in test_url_patterns
    test_url_patterns(pattern.url_patterns, pattern.namespace)
  File &quot;/mnt/magnet/repos/nstuardod-dpt/base/tests.py&quot;, line 231, in test_url_patterns
    self.assertIn(
  File &quot;/usr/lib/python3.8/unittest/case.py&quot;, line 1179, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
  File &quot;/usr/lib/python3.8/unittest/case.py&quot;, line 753, in fail
    raise self.failureException(msg)
AssertionError: 404 not found in (200, 302, 403, 405) : url &quot;/__debug__/render_panel/&quot; (render_panel)returned 404
--------------------------------------- Captured stdout setup ---------------------------------------
Installed 362 object(s) from 1 fixture(s)

--------------------------------------- Captured stderr setup ---------------------------------------
Creating test database for alias &apos;default&apos;...
<font color="#C4A000">========================================= warnings summary ==========================================</font>
base/tests.py::UrlsTest::test_responses
  /home/wooo/.cache/pypoetry/virtualenvs/django-project-template-8zXZtD0D-py3.8/lib/python3.8/site-packages/pypugjs/runtime.py:12: DeprecationWarning: Using or importing the ABCs from &apos;collections&apos; instead of from &apos;collections.abc&apos; is deprecated since Python 3.3, and in 3.10 it will stop working
    from collections import Mapping as MappingType

-- Docs: https://docs.pytest.org/en/stable/warnings.html
------------ generated xml file: /mnt/magnet/repos/nstuardod-dpt/test-results/pytest.xml ------------

====================================== short test summary info ======================================
FAILED base/tests.py::UrlsTest::test_responses - AssertionError: 404 not found in (200, 302, 403, ...
<font color="#CC0000">============================== </font><font color="#CC0000"><b>1 failed</b></font>, <font color="#4E9A06">6 passed</font>, <font color="#C4A000">1 warning</font><font color="#CC0000"> in 3.91s ===============================</font>
</pre>

Pero los antiguos sí pasan... ¿Qué puede ser @nstuardod ?

<pre>&gt; python manage.py test
Creating test database for alias &apos;default&apos;...
Installed 362 object(s) from 1 fixture(s)
System check identified no issues (0 silenced).
.......
----------------------------------------------------------------------
Ran 7 tests in 6.566s

OK
Destroying test database for alias &apos;default&apos;...
</pre>

nstuardod committed:
chore: fix test env detection and avoid debug toolbar URL imports when testing

Due to the way pytest and its plugin loads Django, DEBUG is not overriden
until tests are going to run, making debug toolbar to be imported. Fix that
